Indexing/TermSharing.cpp

```diff
@@ -259,6 +327,8 @@ Literal* TermSharing::insertVariableEquality(Literal* t,unsigned sort)
   if (s == t) {
     t->markShared();
     t->setId(_totalLiterals);
+    //TODO AYB why weight 3? Should the sort be taken into account?
+    //Look at Kernel/LiteralByMatchability
     t->setWeight(3);
```

Comment:
Did you later figure this out?

I'd say "X=Y" has weight 3 simply because of two the variables and one "=". 

Did you experiment, in the HOL setting, with additional weight for the sorts?

Reply:
I have not yet done so. As the comment suggestions, this is likely to cause issues with polymorphic problems in `Kernel::LiteralByMatchability`. Consider the following code:

```
    static unsigned computeRating(Literal* lit)
    {
      return lit->weight() - lit->getDistinctVars();
    }
```

If `lit->weight()` is always three for two variable equalities, this function is going to work as expected. However, such an issue should not affect monomorphic problems.

Reply:
Is it because `lit->getDistinctVars();` will potentially return more than 2 (including variables in the sort)? 

In any case, I understand this is an open issue relevant only for the poly problems.